lib/govspeak/header_extractor.rb

```diff
@@ -3,11 +3,46 @@ module Govspeak
 
   class HeaderExtractor < Kramdown::Converter::Base
     def convert(doc)
+      headers = []
+
       doc.root.children.map do |el|
         if el.type == :header
-          Header.new(el.options[:raw_text], el.options[:level], generate_id(el.options[:raw_text]))
+          headers.push(build_header(el))
+          next
         end
-      end.compact
+
+        headers.push(find_headers(el)) if el.type == :html_element
+      end
+
+      headers.flatten.compact
+    end
+
+    private
```

Comment:
Our style is to indent this at the `class` level and drop the whitespace line beneath. See https://github.com/alphagov/styleguides/blob/master/ruby.md